pkg/stores/proxy/proxy_store.go

```diff
@@ -256,6 +293,14 @@ func (s *Store) list(apiOp *types.APIRequest, schema *types.APISchema, client dy
 
 	tableToList(resultList)
 
+	if resourceVersion := resultList.GetResourceVersion(); resourceVersion != "" {
```

Comment:
I think that's a valid concern.

At the moment, one of the main goals of this cache implementation is to support pagination: whenever visiting a table in the UI, make it so the first page might be as slow to get as today, then make successive pages fast.

The current implementation, as well as this PR, should help in that direction - albeit potentially not fully or in all cases. Hopefully end-to-end testing will clarify just how far this approach can be taken.

We will soon be discussing about a new cache design where this cache works in a very different way, and among other things it could get updates via Watch when resources change. That should more fully address your point. I would love to have you take part in that design discussion if you are open to it.

Reply:
Thanks, I get it now. In this case, I would prefer setting lower timeout for the cache key as the set of cached list for the resource will only be valid in a very short time.
For example, the list will be cached again after the UI page is refreshed.

Reply:
@orangedeng @moio I wonder if there is a reasonable way for us to instead build a key from the `resourceVersion`s of the individual items in the list and hand that back to the requester.

Reply:
Watching resources does sound like the most reasonable way to do caching. It is like a short lifecycle list-watch cache for requested resources.

FYI: I think the current approach is good enough for speeding up partition request. And it would be better if we can clean up caches periodically and set lower timeout for each cache content.